app_test.go

```diff
@@ -944,8 +954,8 @@ func TestApp_SetStdout(t *testing.T) {
 		t.Fatalf("Run error: %s", err)
 	}
 
-	if w.Len() == 0 {
-		t.Error("App did not write output to desired writer.")
+	if w.Len() != 0 {
+		t.Error("App wrote help screen to the wrong stream.")
```

Comment:
The negation of this test case makes me think it's no longer preserving the original intent. Before this change, the assertion was that the writer was written to, whereas now, the only thing we know is that a writer is _not_ written to. 

The naming on this test isn't great here, but I think the change that preserves the functionality tested would be to define the app like so:

```go
	app := &App{
		Name:   "test",
		ErrWriter: &w,
	}
```

And then keep the original assertion

Reply:
This has been addressed. Please review again @rliebz